WIP: Refactor of cloud game overlord and overworker

sadlil opened the pull request:
Work In Progress.

Fixes #68

giongto35 commented:
@_@ wow, it's a massive change. When it finishes, it gonna be awesome. (y)

sadlil commented:
@giongto35 Hope so. I am still doing the changes. Hopefully, you can start to review some part from now. :) Thanks, man.

giongto35 commented:
Yes I'm checking :P

giongto35 commented on pkg/worker/handlers.go:
```diff
@@ -24,23 +24,23 @@ type Handler struct {
 	overlordHost string
 	// Rooms map : RoomID -> Room
-	rooms map[string]*room.Room
+	rooms map[string]*room2.Room
```
Why do you rename it to room2 and storage2?

sadlil replied:
Ah. None. It was my refactoring tool. Will fix it. Thanks.

giongto35 replied:
What is your tool for lint. It's so magic. Teach me senpai.

sadlil replied:
I normally use all the common go tools -
For lint and format:
```
- go fmt
- go import
- golangci-lint
```

For moving packages around I use `gomvpkg` from - https://github.com/golang/tools/tree/master/cmd

And for statci code check i use staticcheck. You will find the installations - `./hack/scripts/install_tools.sh`. And how to uses in the `Makefiles`.

giongto35 commented:
I have a new update on encoder. You may need to introduce new files.
https://github.com/giongto35/cloud-game/pull/70/files

added files:
> encoder/type.go
> h264encoder/encoder.go

sadlil commented:
@giongto35 Go ahead and merge, please. I will fix the conflicts in my branch. :)

sadlil commented:
@giongto35 I have fixed all the merge issues. But while running My game is failing as my `nanoarc.go:L345` is failing to find the game. I have moved the emulators to `assets/emulator/libretro/cores` directory as I think this is more appropriate palace. Can help me with this. What is this url represents? Is it relative to something? Or I need to put the .so files in the same directory?

sadlil commented:
To me the problem looks like a Mac issue. Did you happen to encounter this problem before? Let me know. @giongto35

sadlil commented:
@giongto35 I have completed the first version of refactoring the code.  Apart from the issue with `C.dlopen` everything else works fine. Please have a look. Help fix the error and if its looks okay - lets merge it. For reference, we need to fix the following in future iterations for refactoring

1. Use flags embedded in binary instead of global flags
2. Few more code refactoring inside overlord and worker codes
3. Rewrite Readme
4. Organize and fix docker files and builds
5. Get rid of `pkg/config` and reorganize `pkg/util`
6. Write E2e tests and move `main_test.go` to `tests/e2e` using ginkgo - example - https://github.com/appscode/voyager/blob/master/test/e2e/e2e.go

giongto35 commented:
Sorry @sadlil, I have a problem of github notification, so I forgot to check. :D
This is massive and necessary. I 'm not aware of the problem with Mac because I develop it on Ubuntu. You re so good figuring it out.
And I prioritize your merge, Hopefully to get it merged today or tomorrow. Will not spoil any conflict :p

sadlil commented:
@giongto35 It okay. No worries. Take your time to review and please have a short testing session to check everything works well with this patch.

giongto35 commented:
FYI, vendor removed all C files related to x264 :( . I need to use https://github.com/goware/modvendor to get it back.
BTW, I saw you using vendor but don't have any script based on vendor. I think vendor can only help when we build with `-mod=vendor`

giongto35 commented:
I love the new log <3
